ts/components/wallet/card/CardComponent.tsx

```diff
@@ -186,27 +185,13 @@ export default class CardComponent extends React.Component<Props> {
   }
 
   private renderBody(creditCard: CreditCard) {
-    const { type, wallet } = this.props;
+    const { type } = this.props;
 
     const getBodyIcon = () => {
-      if (
-        this.props.type === "Picking" ||
-        (this.props.type === "Full" && this.props.showPsp)
-      ) {
-        return wallet.psp ? (
-          <View style={[styles.cardPsp]}>
-            <Logo
-              item={creditCard}
-              pspLogo={wallet.psp.logoPSP}
-              imageStyle={styles.pspLogo}
-            />
-          </View>
-        ) : (
-          <View style={[styles.cardPsp]}>
-            <Logo />
-          </View>
-        );
+      if (type === "Picking" || type === "Full") {
```

Comment:
What do you think about using a switch? 
It's safer, and prevent unexpected behaviour if in future we add a type (if you don't use the default of course).
```suggestion
      switch (type) {
        case "Picking":
        case "Full":
          return null;
        case "Header":
        case "Preview":
          return (
            <View style={[styles.cardLogo, { alignSelf: "flex-end" }]}>
              <Logo item={creditCard} />
            </View>
          );
      }
```